src/org/traccar/protocol/OwnTracksProtocolDecoder.java

```diff
@@ -70,34 +114,103 @@ protected Object decode(
         if (root.containsKey("batt")) {
-            position.set(Position.KEY_BATTERY, root.getInt("batt"));
+            position.set(Position.KEY_BATTERY_LEVEL, root.getInt("batt"));
         }
-
-        position.setTime(new Date(root.getJsonNumber("tst").longValue() * 1000));
-
-        String uniqueId;
-
-        if (root.containsKey("topic")) {
-            uniqueId = root.getString("topic");
-            if (root.containsKey("tid")) {
-                position.set("tid", root.getString("tid"));
-            }
-        } else {
-            uniqueId = root.getString("tid");
+        if (root.containsKey("uext")) {
+            position.set(Position.KEY_POWER, root.getJsonNumber("uext").doubleValue());
+        }
+        if (root.containsKey("ubatt")) {
+            position.set(Position.KEY_BATTERY, root.getJsonNumber("ubatt").doubleValue());
+        }
+        if (root.containsKey("vin")) {
+            position.set(Position.KEY_VIN, root.getString("vin"));
+        }
+        if (root.containsKey("name")) {
+            position.set(Position.KEY_VIN, root.getString("name"));
+        }
+        if (root.containsKey("rpm")) {
+            position.set(Position.KEY_RPM, root.getInt("rpm"));
+        }
+        if (root.containsKey("ign")) {
+            position.set(Position.KEY_IGNITION, root.getBoolean("ign"));
+        }
+        if (root.containsKey("motion")) {
+            position.set(Position.KEY_MOTION, root.getBoolean("motion"));
+        }
+        if (root.containsKey("odometer")) {
+            position.set(Position.KEY_ODOMETER, root.getJsonNumber("odometer").doubleValue() * 1000.0);
+        }
+        if (root.containsKey("hmc")) {
+            position.set(Position.KEY_HOURS, root.getJsonNumber("hmc").doubleValue() / 3600.0);
         }
 
-        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, uniqueId);
-        if (deviceSession == null) {
-            sendResponse(channel, HttpResponseStatus.BAD_REQUEST);
-            return null;
+        if (root.containsKey("anum")) {
+            Integer anum = root.getInt("anum");
+            for (Integer i = 0; i < anum; i++) {
+                String indexString = String.format("%02d", i);
+                if (root.containsKey("adda-" + indexString)) {
+                    position.set(Position.PREFIX_ADC + (i + 1), root.getString("adda-" + indexString));
+                }
+                if (root.containsKey("temp_c-" + indexString)) {
+                    position.set(Position.PREFIX_TEMP + (i + 1),
+                        root.getJsonNumber("temp_c-" + indexString).doubleValue());
+                }
+            }
         }
-        position.setDeviceId(deviceSession.getDeviceId());
 
         sendResponse(channel, HttpResponseStatus.OK);
         return position;
     }
 
+    private void setEventOrAlarm(Position position, String t, Integer rty) {
+        if (t.equals("9")) {
+            position.set(Position.KEY_EVENT, Event.TYPE_ALARM);
```

Comment:
This is incorrect. This is a different event.

Reply:
There is no specific event type for it. All device events are of alarm type. You should not set `KEY_EVENT` in the decoder.